models/project_repository.go

```diff
@@ -0,0 +1,188 @@
+package models
```

Comment:
Problem is that models imports application (for implementing interfaces), an application imports project (because it's in the interface). I am very open to discuss the package hierarchy, but I think we should do that as a clean up pass that fixes it everywhere.